### ISSUE TITLE
Reproducing Chapter 1:10 slide code matches on the whole sentence

### ISSUE THREAD

**Issue by jtrudel** [ines/spacy-course#32]
When I run the following code in spaCy  2.0.12:

```
import spacy
from spacy.matcher import Matcher

nlp = spacy.load('en_core_web_sm')

matcher = Matcher(nlp.vocab)
pattern = [{'TEXT': 'iPhone'}, {'TEXT': 'X'}]
matcher.add('IPHONE_PATTERN', None, pattern)

doc = nlp("New iPhone X release date leaked")
matches = matcher(doc)

for match_id, start, end in matches:
    matched_span = doc[start:end]
    print(matched_span.text)
```

I get:
```
New
iPhone
X
release
date
leaked
.
```

I somehow match on every individual word of doc instead of just 'iPhone X'. Not sure why it is.

**Comment by ines** [comment]
Hi! The course was built for spaCy v2.1.x, so you'll need the most recent version to make sure all of the examples run as expected.

The `TEXT` key on the patterns (instead of `ORTH`) was added in v2.1.0. It was actually directly inspired by this course, because I realised that explaining why it's called `ORTH` was unnecessarily complicated and you should just be able to specify `TEXT` for `token.text` instead. Similarly, when creating a `Span` object, the `label` argument can now also be a string instead of only an integer ID. 

Also, to give you some background on what's happening in your case: Because spaCy v2.0.x doesn't know the `TEXT` key, it ignores it. So each dict in the pattern is interpreted as `{}`, i.e. "any token".

**Comment by jtrudel** [comment]
That makes a lot of sense. Thank you for the quick reply!

**Comment by abitofalchemy** [comment]
I got spacy installed and got the same error above. I updated to 2.2.2 and that breaks everything for me. I get error `error: bad escape \p at position 257` when I do:  
```
In [5]: nlp = spacy.load('en_core_web_sm')
---------------------------------------------------------------------------
error                                     Traceback (most recent call last)
<ipython-input-5-12102ff9e1a9> in <module>
----> 1 nlp = spacy.load('en_core_web_sm')
```

I have also downloaded this model: `en_core_web_sm-2.2.0.tar.gz` for manual install. So, my question is how do I get this fixed again?

**Comment by abitofalchemy** [comment]
Update: my system is now fixed, above I had mention the model with 2.2.0, but if you have model version 2.0.0, then you get an error. With the matching version of spacy and the `en_core_web_sm` model things work fine.
